### PR TITLE
Create BlockActivateEvent

### PR DESCRIPTION
**Pull request by Samueljh1**
Adds a new event that is sent when a block is activated. Eg: Flint and Steel -> TNT Block
- Also added pull request to Block Class which adds the event caller

### CODE REVIEW

**Comment by xpyctum** [comment]
Bad practice.

**Comment by Samueljh1** [comment]
why?

**Comment by xpyctum** [comment]
It is not and executable php file

**Comment by Samueljh1** [comment]
@xpyctum Whoops, fixed.

**Comment by BlackShadow1** [comment]
You must create a BlockActivateEvent.php, not a BlockActivateEvent without extension

File: src/pocketmine/event/block/BlockActivateEvent.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	/** @var \pocketmine\block\Block */
+	protected $block;
+	
+	/** @var \pocketmine\item\Item */
+	protected $item;
+
+	/** @var \pocketmine\Player */
+	protected $player;
+
+	public function __construct(Block $block, Item $item, Player $player){
```
Comment (xpyctum) [inline src/pocketmine/event/block/BlockActivateEvent.php]: Maybe $player = null? 
Because caller can do this. (Block -> onActivate)

**Comment by Samueljh1** [comment]
@xpyctum updated.

**Comment by PEMapModder** [comment]
From what I remember, null is not a valid object in PHP to be passed into hinted parameters.

**Comment by Samueljh1** [comment]
@PEMapModder really? it works fine.

**Comment by PEMapModder** [comment]
never mind, just tested passing null works.

**Comment by matcracker** [comment]
It's normal that doesn't work xD You don't call the event

**Comment by thebigsmileXD** [comment]
Create even't but never call it?

**Comment by matcracker** [comment]
It's the same that I said, if the event isn't call in the code, it doesn't work

**Comment by Samueljh1** [comment]
@thebigsmileXD I even referenced the call PR: #3882

**Comment by PEMapModder** [comment]
@Samueljh1 if one pull request is dependent on another, you must make them in the same pull request.

**Comment by PEMapModder** [comment]
As this pull request alone is not meaningful, this is being closed.

**Pull request closed without merging**